lib/submit-request.js

```diff
@@ -185,6 +185,38 @@ SubmitRequest.prototype.commit = function(callback) {
   });
 };
 
+SubmitRequest.prototype._metadataProjection = function() {
+  var request = this;
+
+  // Default behavior
+  if (!request.opMetadataProjection) {
+    return undefined;
+  }
+
+  // Granular projection
+  if (typeof request.opMetadataProjection === 'object') {
+    return request._granularMetadataProjection();
+  }
+
+  // Full projection
+  return request.op.m;
```

Comment:
Did you specifically need the ability to get all metadata?

If it's not something you need at the moment, I'd leave this out for now, returning `undefined` as the fallback case.

Requesting just what's needed is better, so if we leave out a "get-everything" option, that removes the temptation to request all the metadata.